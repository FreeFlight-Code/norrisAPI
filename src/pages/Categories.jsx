import React from "react";
import {connect} from 'react-redux';

class Categories extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      categories: []
    };
  }

  componentDidMount() {
    fetch("https://api.chucknorris.io/jokes/categories")
      .then(res => res.json())
      .then(categories => this.setState({ categories: categories }));
  }

  categoriesList() {
    const { categories } = this.state;
    return categories.map((category, i) => {
      return (
        <div
          className="category"
          onClick={e => {
            this.getJoke(category);
          }}
          key={`category-${i}`}
        >
          {category}
        </div>
      );
    });
  }

  getJoke(category) {
    if (category) {
      fetch(`https://api.chucknorris.io/jokes/random?category=${category}`)
        .then(res => res.json())
        .then(joke => {
          this.props.dispatch({
            type: "DISPLAY_MODAL",
            message: joke.value,
            messageType: "info"
          });
        });
    }
  }

  render() {
    return (
      <div className="page categories">
        <div className="categoryList">{this.categoriesList()}</div>
        {this.getJoke()}
      </div>
    );
  }
}

export default connect()(Categories);
