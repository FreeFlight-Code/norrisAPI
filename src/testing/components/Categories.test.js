import React from 'react';
import Categories from '../../pages/Categories';
import renderer from 'react-test-renderer';

describe("renders Categories", ()=>{
	it('without error', () => {
	const tree = renderer
		.create( < Categories /> ).toJSON();
	expect(tree).toMatchSnapshot();
	})
});