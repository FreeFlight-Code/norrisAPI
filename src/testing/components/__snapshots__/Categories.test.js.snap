// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Categories without error 1`] = `
<div
  className="page categories"
>
  <div
    className="categoryList"
  />
</div>
`;
